modules/chip/ChipInput.tsx

```diff
@@ -1,102 +1,120 @@
 import { IProps as IChipListProps } from "./ChipList";
 
-// @ts-ignore
-export interface IProps extends StandardProps<"input"> {
-  value: string[];
+export interface IProps extends React.ComponentProps<typeof ErrorTextInput> {
+  chips: string[];
   allowDuplicates?: boolean;
-  onChange?: (newList: string[]) => void;
+  onChangeChips: (newList: string[]) => void;
 }
 
 const styleSheet: ThemeStyleSheetFactory = theme => ({
   chipInput: {
-    paddingRight: `${theme.spaceUnit.xs}`,
-    paddingLeft: `${theme.spaceUnit.xs}`,
-    borderRadius: 4,
-    border: `1px solid ${theme.colors.grey.grey50}`,
-    display: "flex",
-    ":focus": {
-      backgroundColor: "red"
-    }
+    display: "flex"
   },
-  input: {
-    border: 0,
-    marginLeft: theme.spaceUnit.xs,
-    ...theme.typography.descriptionMedium,
-    flex: 1,
-    minWidth: 80,
-    outline: "none",
-    minHeight: 40
+  input: {},
+  chipList: {
+    width: "100%"
+  },
+  chipContainer: {
+    width: "100%",
+    margin: 0
+  },
+  chip: {
+    justifyContent: "space-between"
+  }
+});
+
+const inputStyleSheet: ThemeStyleSheetFactory = theme => ({
+  prefixIcon: {
+    width: "100%",
+    marginRight: 0
   }
 });
 
+const ErrorTextInputStyle = ErrorTextInput.extendStyles(inputStyleSheet);
+
 function ChipInput({
   styles,
   cx,
   className,
-  value,
-  onChange,
+  chips,
+  onChangeChips,
   allowDuplicates = false,
   wrappedRef,
   parentStylesheet,
+  value,
   ...props
 }: IProps & WithStylesProps) {
-  const [_list, setList] = useState(value || []);
-  const [inputValue, setInputValue] = useState("");
+  const [inputValue, setInputValue] = useState(value?.toString() ?? "");
 
   const ChipListStyle = useRegistryWithStyles<IChipListProps<string>>(
     "ChipList",
     styleSheet
   );
 
   useEffect(() => {
-    setList(value || []);
+    setInputValue(value?.toString() ?? "");
   }, [value]);
 
   const handleInput = useCallback(
     key => {
       if (key.key === "Enter" && inputValue.trim() !== "") {
-        const isIncluded = _list.includes(inputValue.trim());
+        const isIncluded = chips.includes(inputValue.trim());
         if (allowDuplicates || !isIncluded) {
-          const newList = _list.concat([inputValue.trim()]);
-          setList(newList);
-
-          if (onChange) {
-            onChange(newList);
+          const newList = chips.concat([inputValue.trim()]);
+          if (onChangeChips) {
+            onChangeChips(newList);
           }
         }
         setInputValue("");
       }
     },
-    [_list, allowDuplicates, inputValue, onChange]
+    [chips, allowDuplicates, inputValue, onChangeChips]
   );
 
   const handleChange = useCallback(
     newList => {
-      setList(newList);
-      if (onChange) {
-        onChange(newList);
+      if (onChangeChips) {
+        onChangeChips(newList);
       }
     },
-    [onChange]
+    [onChangeChips]
+  );
+
+  const handleChipClick = useCallback(
```

Comment:
I find this name misleading, as I assume we could in the future allow an action to happen if the user clicks on the actual chip. IMO a clearer name would be handleChipClose 💭 

And the same applies to the ChipList prop and even the CloseableChip prop

Reply:
This is the chip click event, not the close one. This has a different, deletes the chip but sets the text on the input. When closing the input is blanked.